config/base/derive/src/lib.rs

```diff
@@ -1,629 +1,38 @@
-        })
-    }
+    let ast = syn::parse_macro_input!(input as utils::StructWithFields);
+    configurable::impl_configurable(&ast)
 }
 
-// Recreate struct
-impl ToTokens for ViewInput {
-    fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
-        let ViewInput {
-            attrs,
-            vis,
-            ident,
-            generics,
-            fields,
-            ..
-        } = self;
-        let stream = quote! {
-            #(#attrs)*
-            #vis struct #ident #generics {
-                #(#fields),*
-            }
-        };
-        tokens.extend(stream);
-    }
-}
-
-/// Keywords used inside `#[view(...)]`
-mod kw {
-    syn::custom_keyword!(ignore);
-    syn::custom_keyword!(into);
-}
-
-/// Structure to parse `#[view(...)]` attributes
-/// [`Inner`] is responsible for parsing attribute arguments
-struct View<Inner: Parse>(std::marker::PhantomData<Inner>);
-
-impl<Inner: Parse> View<Inner> {
-    fn parse(attr: &Attribute) -> syn::Result<Inner> {
-        attr.path
-            .is_ident("view")
-            .then(|| attr.parse_args::<Inner>())
-            .map_or_else(
-                || {
-                    Err(syn::Error::new_spanned(
-                        attr,
-                        "Attribute must be in form #[view...]",
-                    ))
-                },
-                |inner| inner,
-            )
-    }
-}
-
-struct ViewIgnore {
-    _kw: kw::ignore,
-}
-
-impl Parse for ViewIgnore {
-    fn parse(input: ParseStream) -> syn::Result<Self> {
-        Ok(Self {
-            _kw: input.parse()?,
-        })
-    }
-}
-
-struct ViewFieldType {
-    _kw: kw::into,
-    _eq: Token![=],
-    ty: Type,
-}
-
-impl Parse for ViewFieldType {
-    fn parse(input: ParseStream) -> syn::Result<Self> {
-        Ok(Self {
-            _kw: input.parse()?,
-            _eq: input.parse()?,
-            ty: input.parse()?,
-        })
-    }
-}
-
-impl From<ViewFieldType> for Type {
-    fn from(value: ViewFieldType) -> Self {
-        value.ty
-    }
+/// Derive for config querying and setting. More details in `iroha_config_base` reexport
+#[proc_macro_derive(Configuration)]
+pub fn configuration_derive(input: TokenStream) -> TokenStream {
+    let ast = syn::parse_macro_input!(input as utils::StructWithFields);
+    configuration::impl_configuration(&ast)
 }
 
 /// Generate view for given struct and convert from type to its view.
 /// More details in `iroha_config_base` reexport.
 #[proc_macro]
 pub fn view(input: TokenStream) -> TokenStream {
-    let ast = syn::parse_macro_input!(input as ViewInput);
-    let original = gen_original_struct(ast.clone());
-    let view = gen_view_struct(ast);
-    let impl_from = gen_impl_from(&original, &view);
-    let impl_default = gen_impl_default(&original, &view);
-    let impl_has_view = gen_impl_has_view(&original);
-    let assertions = gen_assertions(&view);
+    let ast = syn::parse_macro_input!(input as utils::StructWithFields);
```

Comment:
I think we shoould keep related logic closer together, so consider 2 options: 
1. Add `view::impl_view` function;
2. Or place all macroses into there related files and then `pub use` them.